service/history/workflow/mutable_state_impl.go

```diff
@@ -4912,6 +4911,35 @@ func (ms *MutableStateImpl) RecordLastActivityStarted(ai *persistencespb.Activit
 	})
 }
 
+func (ms *MutableStateImpl) RegenerateActivityRetryTask(ai *persistencespb.ActivityInfo) error {
```

Comment:
nit: probably rename this to something else, it's more than just generating a task.

Reply:
The goal is to (re)generate retry task. I don't have a better name for that, but open for proposal